### PR TITLE
feature/SPRIND-60

### PR DESCRIPTION
**Pull request by BtencateSphereon**

### CODE REVIEW

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks

File: packages/identifier-resolution/src/agent/IdentifierResolution.ts [hunk 1]
```diff
@@ -106,9 +106,16 @@ export class IdentifierResolution implements IAgentPlugin {
     return (await this.identifierGetManaged({ ...args, method: 'cose_key' }, context)) as ManagedIdentifierCoseKeyResult
   }
 
+  private async identifierGetManagedByIssuer(
```
[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/identifier-resolution/src/agent/IdentifierResolution.ts]: renamed all of them to prefix OID4VC

File: packages/identifier-resolution/src/functions/managedIdentifierFunctions.ts [hunk 2]
```diff
@@ -270,6 +273,41 @@ export async function getManagedX5cIdentifier(
...
+export async function getManagedIssuerIdentifier(
+    opts: ManagedIdentifierIssuerOpts,
+    context: IAgentContext<IKeyManager>
+): Promise<ManagedIdentifierIssuerResult> {
+  const { identifier } = opts
+  const method = 'oid4vci-issuer'
+  // FIXME: We need to eventually determine the JWK based on the issuer. Using a dummy JWK for now
+  const jwk = {
+    "kty" : "RSA",
+    "kid" : "cc34c0a0-bd5a-4a3c-a50d-a2a7db7643df",
```
[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/identifier-resolution/src/functions/managedIdentifierFunctions.ts]: changed value to dummy-jwk-for-vci-issuer-signing

File: packages/identifier-resolution/src/types/managedIdentifierTypes.ts [hunk 3]
```diff
@@ -102,7 +122,7 @@ export interface ManagedJwkInfo extends JwkInfo {
 export interface IManagedIdentifierResultBase extends ManagedJwkInfo {
   method: ManagedIdentifierMethod
   opts: ManagedIdentifierOpts
-  key: IKey
+  key?: IKey
```
[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/identifier-resolution/src/types/managedIdentifierTypes.ts]: i thought we discussed to make it optional. changed it to add a dummy IKey

File: packages/identifier-resolution/src/types/managedIdentifierTypes.ts [hunk 4]
```diff
@@ -130,10 +150,6 @@ export function isManagedIdentifierKeyResult(object: IManagedIdentifierResultBas
   return object!! && typeof object === 'object' && 'method' in object && object.method === 'key'
 }
 
-export function isManagedIdentifierCoseKeyResult(object: IManagedIdentifierResultBase): object is ManagedIdentifierCoseKeyResult {
```
[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/identifier-resolution/src/types/managedIdentifierTypes.ts:133]: mistake, should not have been removed. added back

File: packages/identifier-resolution/src/types/managedIdentifierTypes.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nklomp) [reply packages/identifier-resolution/src/types/managedIdentifierTypes.ts]: Yeah, but that was before we discussed the part of in the future having VCI Issuer resolution of keys. Just wanted to know the impact of it. But having a dummy for now like the JWK is fine

**Review by nklomp — APPROVED** [review]

**Comment by nklomp** [comment]
LGTM

**Pull request merged**